src/ui/components/filters/facetscomponent.js

```diff
@@ -252,13 +252,37 @@ export default class FacetsComponent extends Component {
       updatedFacets = previousFacets;
     }
 
+    updatedFacets = this._transformBooleanFacets(updatedFacets);
```

Comment:
This is a minor thing, but it feels a bit weird that we pass all facets through a method called `_transformBooleanFacets`, even if they may not be boolean facets at all. What would you think about having a method called `processFacets` or something like that. The method could then have some logic like:
```
processFacets(facets) {
   // Find facets that are boolean facets, parse only those with _transformBooleanFacets.
   // If any other specific type of facet needs special processing in the future, this would be the
  // entry point for that tool.
}
```

Reply:
I like that idea for the code organization. The one thing I want to be careful of though is not looping through the facets too many times, so I'm going to see if I can do it in one pass.